monkey/infection_monkey/exploit/powershell.py

```diff
@@ -147,57 +161,42 @@ def _report_login_attempt(self, result: bool, credentials: Credentials):
         else:
             raise ValueError(f"Unknown secret type {credentials.secret_type}")
 
-    def _execute_monkey_agent_on_victim(self) -> bool:
-        arch = self._client.get_host_architecture()
-        self.is_32bit = arch == WIN_ARCH_32
-        logger.debug(f"Host architecture is {arch}")
+    def _execute_monkey_agent_on_victim(self):
+        monkey_path_on_victim = self.options["dropper_target_path_win_64"]
 
-        monkey_path_on_victim = (
-            self._config.dropper_target_path_win_32
-            if self.is_32bit
-            else self._config.dropper_target_path_win_64
-        )
+        self._copy_monkey_binary_to_victim(monkey_path_on_victim)
+        logger.info("Successfully copied the monkey binary to the victim.")
 
-        is_monkey_copy_successful = self._copy_monkey_binary_to_victim(monkey_path_on_victim)
-        if is_monkey_copy_successful:
-            logger.info("Successfully copied the monkey binary to the victim.")
+        try:
             self._run_monkey_executable_on_victim(monkey_path_on_victim)
-        else:
-            logger.error("Failed to copy the monkey binary to the victim.")
-            return False
+        except Exception as ex:
+            raise RemoteAgentExecutionError(
+                f"Failed to execute the agent binary on the victim: {ex}"
+            )
 
-        return True
+    def _copy_monkey_binary_to_victim(self, monkey_path_on_victim):
 
-    def _copy_monkey_binary_to_victim(self, monkey_path_on_victim) -> bool:
-        try:
-            self._write_virtual_file_to_local_path()
+        temp_monkey_binary_filepath = f"monkey_temp_bin_{get_random_file_suffix()}"
+
+        self._create_local_agent_file(temp_monkey_binary_filepath)
 
+        try:
             logger.info(f"Attempting to copy the monkey agent binary to {self.host.ip_addr}")
-            is_monkey_copy_successful = self._client.copy_file(
-                TEMP_MONKEY_BINARY_FILEPATH, monkey_path_on_victim
-            )
+            self._client.copy_file(temp_monkey_binary_filepath, monkey_path_on_victim)
         except Exception as ex:
-            raise ex
+            raise RemoteAgentCopyError(f"Failed to copy the agent binary to the victim: {ex}")
```

Comment:
Where is this caught?

Reply:
On line 84. It doesn't need any special processing by the handler. Reraising as a different exception allows us to add more context. We could also reraise as a generic `Exception`, but this should be ok for now.